draft-ietf-mls-protocol.md

```diff
@@ -2532,14 +2532,32 @@ uses:
    regard to any removed members and PCS for the committer and any updated
    members.
 
+When creating or processing a Commit, three different ratchet trees and
+their associated GroupContexts are used:
+
+1. "Old" refers to the ratchet tree and GroupContext for the epoch before the
+   commit.  The old GroupContext is used when signing the MLSPlainText so that
+   existing group members can verify the signature before processing the
+   commit.
+2. "Provisional" refers to the ratchet tree and GroupContext constructed after
+   applying the proposals that are referenced by the Commit.  The provisional
+   GroupContext uses the epoch number for the new epoch, and the old confirmed
+   transcript hash.  This is used when creating the UpdatePath, if the
+   UpdatePath is needed.
+3. "New" refers to the ratchet tree and GroupContext constructed after applying
```

Comment:
I'm still struggling with the fact that there are 3 ratcheting trees. In my mind there are only 2:
 - The ratcheting tree *before* applying proposals and the (potential) UpdatePath
 - The ratcheting tree *after* applying proposals and the (potential) UpdatePath

The ratcheting trees described here as "provisional" and "new" are in fact identical. It's just that they exist at different times. I think it would be good to make that clear, i.e. a "provisional" ratcheting tree becomes a "new" ratcheting tree when a Commit is applied.

Reply:
The "provisional" ratchet tree is the tree after applying the proposals, but before applying the UpdatePath.  We could probably do without a "provisional" ratchet tree, but I think it helps with clarity to give it a name since applying the proposals and applying the UpdatePath are different steps.

Reply:
Discussed at 20211004 MLS interim.

Reply:
Fine, that makes sense to me!